### PR TITLE
Master branch landing page updates

### PR DESCRIPTION
**Pull request by ajkannan**
The new landing page uses the same template as the other GCP language websites.  These changes reflect the addition of a static html page and associated css/java supporting files.

### CODE REVIEW

**Comment by jgeewax** [comment]
Do you have gh-pages branch of your own for us to preview and give feedback?

**Comment by ajkannan** [comment]
http://ajkannan.github.io/gcloud-java/site/0.0.7/site/index.html

(I just realized I added an extraneous hierarchy level in the URL, so I'll take that out.  But here is the URL for now, and I will update after the fix.)

**Comment by jgeewax** [comment]
Can we change "Quickstart with Maven: Add this to your pom.xml" to "Add gcloud to pom.xml"

I'll open a separate issue for the example to be tabbed (we need to show people auth code if they want that).

File: src/site/resources/index.html [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		    code you have to write. The library provides high-level API
+		    abstractions so they're easier to understand. It embraces
+		    idioms of Java, works well with the standard library, and
+		    integrates better with your codebase.
+		    All this means you spend more time creating code that matters
+		    to you.</p>
+		  
+		  <p><code>gcloud</code> is configured to access Google Cloud Platform
+		    services and authorize (OAuth 2.0) automatically on your behalf.
+		    With a quick install and a private key, you are up and ready
```
Comment (aozarov) [inline src/site/resources/index.html]: I am not sure if "install" is the right description in our case. maybe "setup"/"config"/..?

Comment (aozarov) [inline src/site/resources/index.html]: "a private key" - I see that this is taken from other documentation (@jgeewax feedback on it would be great) but
I wonder if that is what we should say in at this "introductory section".  Consider this scenarious:
1) AE - No need for key (auto-detected).
2) Compute engine - No need for key (auto-detected).
3) Other - Use key or authenticated using gcloud sdk.

Indeed a key could be use in all scenarios but not mandatory in either case...

File: src/site/resources/index.html [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		    to go. Better yet, if you are running on a Google Compute Engine
+		    instance, the installation is enough!</p>
```
Comment (aozarov) [inline src/site/resources/index.html]: "installation" - see above.

File: src/site/resources/index.html (hunk 1 above)
Comment (jgeewax) [inline src/site/resources/index.html]: I'm also not sure about "install" as the word for Java people. I think we're copying mostly from https://googlecloudplatform.github.io/gcloud-node/#/ which is fine... Though the example should be where we call out that GAE/GCE *can* auto-detect credentials for you, but regardless of where you are, you can set credentials manually.

@mziccard, thoughts? What do you call "adding the lines to your pom file" ? "Installing"? or something else?

File: src/site/resources/index.html [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+		  
+                  <div hljs="" language="js">
+		    <pre><code class="hljs java"><span class="hljs-keyword">import</span> com.google.gcloud.datastore.Datastore;
+<span class="hljs-keyword">import</span> com.google.gcloud.datastore.DatastoreFactory;
+<span class="hljs-keyword">import</span> com.google.gcloud.datastore.DatastoreOptions;
+<span class="hljs-keyword">import</span> com.google.gcloud.datastore.Entity;
+<span class="hljs-keyword">import</span> com.google.gcloud.datastore.Key;
+<span class="hljs-keyword">import</span> com.google.gcloud.datastore.KeyFactory;
+
+DatastoreOptions options = DatastoreOptions.builder().projectId(<span class="hljs-string">PROJECT_ID</span>).build();
```
Comment (aozarov) [inline src/site/resources/index.html]: projectId is required but auto-detected in CE and AE or when logged in via gcloud SDK. Not sure what is the right thing to do about it:
- leave as is
- add a comment that may not be required in ...
- remove (and maybe add a comment that need to be specified when if could not auto-detected)

@jgeewax  ?

Comment (jgeewax) [inline src/site/resources/index.html]: This is why we would have a tabbed view like the gcloud-node landing page. I say we leave this for now, and change it a bit later.

File: src/site/resources/index.html (hunk 1 above)
Comment (mziccard) [inline src/site/resources/index.html]: Usually in `maven`/`gradle`/`sbt` you *add a dependency* and *install a plugin*. I would rephrase to something like:
> Add `gcloud` dependency to you project and get a private key to be up and ready to go. Better yet, if you are running on a Google Compute Engine instance, the private key is automatically detected.

Comment (mziccard) [inline src/site/resources/index.html]: Integrating my previous comment:
To add the dependency with `gradle` add it to the `build.gradle` file:
```
dependencies {
  // ...
  compile group: 'com.google.gcloud', name: 'gcloud-java', version: '0.0.6'
}
``` 
With sbt add to the `build.sbt` file:
```
libraryDependencies += "com.google.gcloud" % "gcloud-java" % "0.0.6"
'''

**Comment by ajkannan** [comment]
Here is an updated link: http://ajkannan.github.io/gcloud-java/site/0.0.7/index.html

File: src/site/resources/index.html (hunk 3 above)
Comment (aozarov) [inline src/site/resources/index.html]: I see, @ajkannan if you think adding the tabbed view is simple I would say lets do it now otherwise lets create an issue for it (maybe in the case of 'elsewhere' we could add a comment
suggesting `gcloud login` could also be used for authentication [and selecting projectId]).

@jgeewax looks like in gcloud-node you still require project-id on CE or is that doc not up-to-date? Also, I wonder why we only mention CE and not AE?

File: src/site/resources/index.html [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+		    All this means you spend more time creating code that matters
+		    to you.</p>
+		  
+		  <p><code>gcloud</code> is configured to access Google Cloud Platform
+		    services and authorize (OAuth 2.0) automatically on your behalf.
+		    Add the <code>gcloud</code> dependency to your project and get a private key to be
+		    up and ready to go. Better yet, if you are running on a Google
+		    Compute Engine instance, the private key is automatically detected.
+		  
+                  <h4>Retrieve Datastore Entries</h4>
```
Comment (aozarov) [inline src/site/resources/index.html]: "Example: " or "Example of using Cloud Datastore"?

File: src/site/resources/index.html (hunk 3 above)
Comment (jgeewax) [inline src/site/resources/index.html]: Node isn't supported on App Engine outside of Managed VMs. And yes we require a project ID for the other libraries. I believe there's a gcloud-common issue on that topic though.

File: src/site/resources/index.html [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+KeyFactory keyFactory = datastore.newKeyFactory().kind(<span class="hljs-string">KIND</span>);
+Key key = keyFactory.newKey(keyName);
+Entity entity = datastore.get(key);</code></pre></div>
+                </div><!-- end of .container -->
+            </section><!-- end of .featuring -->
+
+            <section class="block faq">
+                <div class="container">
+                    <h3 class="block-title">FAQ</h3>
+		    <h4>What is the relationship between the gcloud package and the gcloud command-line tool?</h4>
```
Comment (aozarov) [inline src/site/resources/index.html]: I think for Java saying "gcloud package" is not as clear as in Python or node where "gcloud" is indeed the package name...

File: src/site/resources/index.html [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+Datastore datastore = DatastoreFactory.instance().get(options);
+KeyFactory keyFactory = datastore.newKeyFactory().kind(<span class="hljs-string">KIND</span>);
+Key key = keyFactory.newKey(keyName);
+Entity entity = datastore.get(key);</code></pre></div>
+                </div><!-- end of .container -->
+            </section><!-- end of .featuring -->
+
+            <section class="block faq">
+                <div class="container">
+                    <h3 class="block-title">FAQ</h3>
```
Comment (aozarov) [inline src/site/resources/index.html]: I am not sure what is the right way to do that by pointing to gcloud-common FAQ could be handy. @jgeewax ?

**Comment by aozarov** [comment]
Looks good to me. @jgeewax ?

File: src/site/resources/index.html (hunk 3 above)
Comment (ajkannan) [inline src/site/resources/index.html]: It looks like I'll need to hunt through dependencies to get the right library for tabs.  I'll file a separate issue for it and work on it after this gets merged.

**Comment by aozarov** [comment]
This is a fix for #66
